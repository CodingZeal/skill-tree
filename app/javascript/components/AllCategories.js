import React from "react";

import SingleCategory from "./SingleCategory";

function AllCategories({ myRatings }) {
  const categoryList = myRatings.map((rating, index) => {
    return <SingleCategory index={index} key={rating.id} rating={rating} />;
  });

  return <div className="allCategories">{categoryList}</div>;
}

export default AllCategories;
