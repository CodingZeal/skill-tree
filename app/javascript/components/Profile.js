import React from "react";

import AllCategories from "./AllCategories";
import { myRatings, oneUser } from "./API/api";

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      myRatings: [],
      user: []
    };
  }

  componentWillMount() {
    this.renderingFunction();
  }

  componentDidUpdate(prevProps) {
    const { match } = this.props;

    if (match.params.id !== prevProps.match.params.id) {
      this.renderingFunction();
    }
  }

  renderingFunction() {
    const { match } = this.props;
    const { id } = match.params;

    oneUser(id).then(APIuser => {
      this.setState({ user: APIuser });
    });
    myRatings(id).then(APIrating => {
      this.setState({ myRatings: APIrating });
    });
  }

  render() {
    const { user, myRatings } = this.state;
    // coming from fetch of profile (find where(url = {url}))
    const host = window.location.origin;

    // local host will change on deployment
    const myUrl = `${host}/profile/`;

    return (
      <div className="profile">
        <h1 className="card-header">My Profile</h1>
        <div className="card">
          <div className="card-content">
            <h1 className="card-info" id="fullname">
              {user.first_name} {user.last_name}
            </h1>
            <h2 className="card-info" id="email">
              <span aria-label="envelope" role="img">
                {" "}
                ‚úâÔ∏è
              </span>{" "}
              {user.email}
            </h2>
            <h2 className="card-info" id="timezone">
              <span aria-label="globe" role="img">
                {" "}
                üåê
              </span>{" "}
              {user.time_zone}
            </h2>
            <h2 className="card-info" id="url">
              {myUrl}
              <br />
              {user.unique_url}
            </h2>
          </div>
          <div className="categories">
            <AllCategories myRatings={myRatings} />
          </div>
        </div>
      </div>
    );
  }
}

export default Profile;
