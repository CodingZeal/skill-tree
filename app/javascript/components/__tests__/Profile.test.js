import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import AllCategories from "../AllCategories";
import Profile from "../Profile";
import "whatwg-fetch";

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe("<Profile>", () => {
  const current_user = {
    id: 1
  };
  const profile = {
    email: "john@example.com",
    firstname: "John",
    id: 1,
    lastname: "Doe",
    timezone: "UTC",
    url: "iou22o38f4"
  };

  const wrapper = shallow(
    <Profile
      match={{ params: { id: 1 } }}
      profile={profile}
      current_user={current_user}
    />
  );

  it("renders corretly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("displays user fullname", () => {
    expect(wrapper.find("#fullname").length).toEqual(1);
  });
  it("displays user email", () => {
    expect(wrapper.find("#email").length).toEqual(1);
  });
  it("displays user timezone", () => {
    expect(wrapper.find("#timezone").length).toEqual(1);
  });
  it("displays a unique url", () => {
    expect(wrapper.find("#url").length).toEqual(1);
  });

  it("should render the AllCategories component", () => {
    expect(wrapper.find(AllCategories)).toHaveLength(1);
  });
});
