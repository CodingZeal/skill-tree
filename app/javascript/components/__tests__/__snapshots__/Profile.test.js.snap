// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> renders corretly 1`] = `
<div
  className="profile"
>
  <h1
    className="card-header"
  >
    My Profile
  </h1>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <h1
        className="card-info"
        id="fullname"
      >
         
      </h1>
      <h2
        className="card-info"
        id="email"
      >
        ‚úâÔ∏è 
      </h2>
      <h2
        className="card-info"
        id="timezone"
      >
        üåê 
      </h2>
      <h2
        className="card-info"
        id="url"
      >
        http://localhost/profile/
        <br />
      </h2>
    </div>
    <div
      className="categories"
    >
      <AllCategories
        myRatings={Array []}
      />
    </div>
  </div>
</div>
`;
